javalib/src/main/scala/java/lang/Character.scala

```diff
@@ -378,7 +378,7 @@ object Character {
         }
         if (isHighSurrogate(seq(i))) {
           val next = i + 1
-          if (next <= end && isLowSurrogate(seq(next))) {
+          if (next < end && isLowSurrogate(seq(next))) {
```

Comment:
Though there shouldn't be a difference, it might be more accurate by

```scala
if (next <= end && next < seq.length && isLowerSurrogate(seq(next))) {
```
?

Reply:
That's a good idea. I also checked the Scala.js's implementation and it seems they don't have an implementation [Character#offsetByCodePoints](https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html#offsetByCodePoints-char:A-int-int-int-int-) and there's only a `String.offsetByCodePoints`.

It might be good to sync after we implement `Character.offsetByCodePoints` on scala.js side :+1: